src/utils/components/withMatchMedia.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { MEDIA_QUERIES } from '../designConstants';
+
+/**
+ * @param {String} breakpoint - name of breakpoint (for example, 'medium')
+ * @returns {String} - state property name (for example, isAtMediumUp)
+ */
+export const getStateNameByBreakpoint = breakpoint => {
+	const capitalizedBp = breakpoint[0].toUpperCase() + breakpoint.slice(1);
+	return `isAt${capitalizedBp}Up`;
+};
+
+/**
+ * @param {Array} mediaQueries - list of matchMedia-created MediaQueryList objects
+ * @param {Array} breakpoints - array of breakpoint names that were passed to HOC
+ */
+export const getUpdatedMediaState = (mediaQueries, breakpoints) => {
+	const updatedMedia = {};
+	mediaQueries.forEach((mq, i) => {
+		updatedMedia[getStateNameByBreakpoint(breakpoints[i])] = mq.matches;
+	});
+	return updatedMedia;
+};
+
+/**
+ * @param {Array} breakpoints
+ * @throws {Error}
+ * @return {undefined}
+ */
+export const validateBreakpoints = (breakpoints) => {
+	const validBreakpoints = Object.keys(MEDIA_QUERIES);
```

Comment:
no need to redefine `validBreakpoints` every time the function runs - make it a module-scoped `VALID_BREAKPOINTS` constant, or curry the function:

```js
// look ma, pure functions!
const _getValidateBreakpoints = mediaQueries => {
  const validBreakpoints = Object.keys(mediaQueries);
  return breakpoints => {
    ...
  }
};

const validateBreakpoints = _getValidateBreakpoints(MEDIA_QUERIES);